docs/proposals/failover/optimize-cluster-failover.md

```diff
@@ -0,0 +1,635 @@
+
+### Non-Goals
+
+1. In the cluster failure detection, provide specific extended implementations for the cluster status detection method.
+
+2. Provide advanced Failover capabilities, such as resource eviction based on strategies like workload priority and remaining resources.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+
+As an administrator or an operation and maintenance personnel, one should be able to explicitly view and judge during daily inspections and maintenance processes:
+
+- At the system configuration level: Whether the cluster failure migration is enabled.
+- At the cluster failure condition judgment level: Which cluster status conditions will mark a cluster as failure, thus triggering the failure migration of resources on the cluster.
+- At the application-level cluster failover strategy: For a specific workload, what is its specific migration strategy when a cluster failure occurs.
+
+#### Story 2
+
+As an application maintenance personnel, one should be able to customize the migration strategy for the cluster failure scenario:
+
+- For general workloads, when a hard failure occurs in the cluster (such failures will affect most of the workloads), trigger the migration of most workloads on the cluster;
+- For workloads with high resource requirements, when a partial failure occurs in the cluster (such failures will only affect a part of the workloads, causing some workloads to be unable to run or their performance to be affected), only trigger the migration of such workloads.
+
+#### Story 3
+
+As a conservative and cautious user, a large number of businesses adopt a fixed redundant deployment scheme, and I don't want any non-human-initiated migrations to occur in the event of a cluster failure. A small number of businesses support a dynamic HA deployment scheme, and I hope that when a cluster failure occurs, the affected parts of these businesses can be migrated to a new healthy cluster.
+
+## Design Details
+
+The overall process of Cluster failover is as follows:
+
+![cluster failover overview](optimize-cluster-failover-overview.png)
+
+Regarding the overall process of Cluster Failover, after re-sorting, it is divided into the following stages:
+1. Input of cluster status conditions.
+2. Matching of cluster failure strategies.
+3. System response to failures (preventing the scheduling of new workloads or triggering workload failure migration).
+
+### Overall Design Principles
+
+Below, we present the overall design principles of Cluster Failover:
+
+1. Judgment of cluster failure conditions
+- The sources for judging cluster failures are diverse. The cluster status detection that Karmada supports by default will first be reflected in `cluster.status.conditions`.
+- After the Karmada system determines that a cluster has a failure, it will add corresponding taints to the cluster object. In cases of severe failures, the migration process will be triggered.
```

Comment:
From the description above, it seems that the Karmada system WILL NOT taint the cluster without user declarative configurations.

Reply:
Yes, it is. The user completely handles the cluster tainting.

Reply:
?

> After the Karmada system determines that a cluster has a failure, it will add corresponding taints to the cluster object.

But this is saying once a cluster becomes a failure, the taints will be added to the cluster.

Reply:
The description of this point is misleading and should be deleted.